ccl/multiregionccl: TestRegionLivenessProber failed

ccl/multiregionccl.TestRegionLivenessProber [failed](https://mesolite.cluster.engflow.com/invocations/default/cce5ad0b-8242-4f58-8306-d9042cf25cc1?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [fed46b2b6c334814209968e98cdc41ed58c691ad](https://github.com/cockroachdb/cockroach/commits/fed46b2b6c334814209968e98cdc41ed58c691ad):


```
=== RUN   TestRegionLivenessProber
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRegionLivenessProber1563638306
    test_log_scope.go:76: use -show-logs to present logs inline
    regionliveness_test.go:176: 
        	Error Trace:	pkg/ccl/multiregionccl/regionliveness_test.go:176
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRegionLivenessProber
        	Messages:   	one region should be dead
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRegionLivenessProber1563638306
--- FAIL: TestRegionLivenessProber (16.90s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProber.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45962

ccl/multiregionccl.TestRegionLivenessProber [failed](https://tanzanite.cluster.engflow.com/invocations/default/c64576be-6cad-4dcd-8b43-75193d3a049f?testReportRun=16&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [bc6d6e05a7c0f9ffd8103740239fdbc83fa78e3f](https://github.com/cockroachdb/cockroach/commits/bc6d6e05a7c0f9ffd8103740239fdbc83fa78e3f):


```
=== RUN   TestRegionLivenessProber
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRegionLivenessProber1917581681
    test_log_scope.go:76: use -show-logs to present logs inline
    regionliveness_test.go:176: 
        	Error Trace:	pkg/ccl/multiregionccl/regionliveness_test.go:176
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRegionLivenessProber
        	Messages:   	one region should be dead
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRegionLivenessProber1917581681
--- FAIL: TestRegionLivenessProber (19.91s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=16</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProber.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/multiregionccl/regionliveness_test.go
+++ b/pkg/ccl/multiregionccl/regionliveness_test.go
@@ -62,13 +62,13 @@ const (
 func TestRegionLivenessProber(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	// Force extra logging to diagnose flakes.
+	require.NoError(t, log.SetVModule("prober=2"))
 	// This test forces the SQL liveness TTL be a small number,
 	// which makes the heartbeats even more critical. Under stress and
 	// race environments this test becomes even more sensitive, if
 	// we can't send heartbeats within 20 seconds.
-	skip.UnderStress(t)
-	skip.UnderRace(t)
-	skip.UnderDeadlock(t)
+	skip.UnderDuress(t)
 
 	ctx := context.Background()
 
@@ -125,11 +125,6 @@ func TestRegionLivenessProber(t *testing.T) {
 		_, err = tenantSQL[0].Exec(fmt.Sprintf("ALTER DATABASE system ADD REGION '%s'", expectedRegions[i]))
 		require.NoError(t, err)
 	}
-	// Override the table timeout probe for testing.
-	for _, ts := range tenants {
-		regionliveness.RegionLivenessProbeTimeout.Override(ctx, &ts.ClusterSettings().SV, testingRegionLivenessProbeTimeout)
-	}
-
 	var cachedRegionProvider *regions.CachedDatabaseRegions
 	cachedRegionProvider, err = regions.NewCachedDatabaseRegions(ctx, tenants[0].DB(), tenants[0].LeaseManager().(*lease.Manager))
 	require.NoError(t, err)

--- a/pkg/sql/regionliveness/BUILD.bazel
+++ b/pkg/sql/regionliveness/BUILD.bazel
@@ -27,6 +27,7 @@ go_library(
         "//pkg/sql/sqlliveness/slbase",
         "//pkg/sql/types",
         "//pkg/util/encoding",
+        "//pkg/util/log",
         "//pkg/util/timeutil",
         "@com_github_cockroachdb_errors//:errors",
     ],

--- a/pkg/sql/regionliveness/prober.go
+++ b/pkg/sql/regionliveness/prober.go
@@ -28,6 +28,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slbase"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 )
@@ -200,6 +201,7 @@ func (l *livenessProber) ProbeLivenessWithPhysicalRegion(
 
 	// Region is alive or we hit some other error.
 	if err == nil || !IsQueryTimeoutErr(err) {
+		log.VEventf(ctx, 2, "region probe completed with error: %v", err)
 		return err
 	}
 
@@ -222,10 +224,12 @@ func (l *livenessProber) ProbeLivenessWithPhysicalRegion(
 		if err != nil {
 			return err
 		}
+		log.VEventf(ctx, 2, "marking region %q as dead at time: %s", string(regionBytes), txnTS.String())
 		if err := txn.Run(ctx, ba); err != nil {
 			// Conditional put failing is fine, since it means someone else
 			// has marked the region as dead.
 			if errors.HasType(err, &kvpb.ConditionFailedError{}) {
+				log.VEventf(ctx, 2, "ignoring condition failed for region: %q", string(regionBytes))
 				return nil
 			}
 			return err

